panorama_connector.py

```diff
@@ -474,9 +476,18 @@ def _get_all_device_groups(self, param, action_result):
         if not result_data:
             return (action_result.set_status(phantom.APP_ERROR, "Got empty list for device groups"), device_groups)
 
+        self.debug_print('Getting Device Groups config from %s' % result_data)
+        device_groups_config = result_data.pop()
+
+        if not isinstance(device_groups_config, dict):
+            error_msg = "Invalid Device Group config: %s" % device_groups_config
+            self.debug_print(error_msg)
+            return action_result.set_status(phantom.APP_ERROR, error_msg), []
+
         try:
-            device_groups = result_data[0]['device-group']['entry']
+            device_groups = device_groups_config['device-group']['entry']
```

Comment:
I have to make change here, because we are supposed to use the latest element (i.e. the response of the API call we've just done within this method).
The issue is that `action_result` is an object that might be modified before this method gets called. 
Therefore, using the 1st element does not always work (i.e. 1st element does not always point to the response from the API call we just did above)